modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java

```diff
@@ -854,6 +876,33 @@ public String getDDMFormFieldDataType() {
 			}
 		}
 
+		public String getDDMFormFieldInputControl() {
+			try {
+				String type = _ddmStructure.getFieldType(
+					_ddmFormFieldValue.getName());
+
+				if (Objects.equals(type, DDMFormFieldType.CHECKBOX) ||
+					Objects.equals(type, DDMFormFieldType.RADIO) ||
+					Objects.equals(type, DDMFormFieldType.SELECT) ||
+					Objects.equals(type, DDMFormFieldType.TEXT) ||
+					Objects.equals(type, DDMFormFieldType.TEXT_AREA)) {
+
+					return type;
+				}
+				else return null;
```

Comment:
ups, this if surely a bug of the source formatter

Reply:
Already fixed by your changes